### ISSUE TITLE
GPS: panic 24,15 ON_TX timeout

### ISSUE THREAD

**Issue by cire831** [MamMark/mm#57]
saw a PANIC_WARN somewhat before the panic.  may not be relevant,  OE on TX.

then saw the panic, ON_TX and timedout.

sending sirf_close_session and somehow timedout.  length is 11 and idx is 8 (when it timedout).

==> something killed the tx interrupt mechanism.

```
gps on A0
A0:   ie:   03   ifg:   0c    iv: 0000  rx:   b3     tx:  00  stat: 09
   ctlw0: 0080    br: 0048  mctl: ee00            n[16]: Epa
```
09 means BRK and UCBUSY.  UCBUSY is probably from the tx being busy.  where did the brk come from?

Now the above is after human time.

we want to instrument the h/w state (usci, timing information, etc.) if a Timer expires.  ie TXTimer.

**Comment by cire831** [comment]
saw again.  tx timeout.

work around would be something like:
- verify we have a tx transmit (how? modify interface)
- send gps_send_block_stop
- abort GPSTransmit.send
- set state to something reasonable
- return error to original caller of GPSTransmit.send (GPSMonitor)
- need to modify GPSTransmit.send_done to return error
- only one caller/user of GPSTransmit (GPSMonitor)
- possible Gsd4eU state machine changes.

**Comment by cire831** [comment]
do as part of GPSMonitor State Machine

**Comment by cire831** [comment]
Exception State:
```
(gdb) excep
exception state
iF 0  control: ---  msp  p    xPSR: nzCvn T  ge: 0   x  exc: 0(d)
shcsr: (00070000) p:    a:     syshandler
(h/c)fsr:  ooo 
bfar:  v: e000edf8    
mmfar: v: e000edf8    
icsr:  (00438000) (vectn.. P: 56(d)  r  0(d) )
nvic: enable:   0041 0c00   0000 01f8
      pending:  0000 0000   0000 0100
      active:   0000 0000   0000 0000
```

Pending above ```0000 0000 0000 0100``` is P6, P6.1 is interrupting and enabled.  P6.1 is radio_irqn (doesn't matter, not relavent)


GPS is on Port A0:  (state at time of exception)
```
(gdb) us
A0:   ie:   03   ifg:   0c    iv: 0000  rx:   b3     tx:  00  stat: 09
   ctlw0: 0080    br: 0048  mctl: ee00            n[16]: Epa
```
Interrupts are enabled (at NVIC level, ```n[16] Epa```),  A0 IE has RX and TX turned on.
IFG 0C -> ```TXCPT``` ```STTIFG``` and ```RXIFG```.  RXIFG is nothing, we are sitting at a breakpoint and the GPS is sending.

Looks like a TXINT went missing or got cleared somehow.
```
GPS0HardwareP__
    m_tx_buf    sirf_go_mpm
    m_tx_len    0xE                    (14)
    m_tx_idx    0xB                    (11)
```

Still more to do....

**Comment by cire831** [comment]
A different 24,15 failure mechanism shows up like:

```
gps(ua0): pwr (g/m): 1  on_off: 0  nRst: 1  awake: 1    grt
                     O          O        O         I     MM
A0:   ie:   03   ifg:   0e    iv: 0004  rx:   e2     tx:  02  stat: 01
   ctlw0: 0080    br: 0048  mctl: ee00            n[16]: EPa

```

iv says tx int is pending but we timed out.  Conjecture is the tx int is being starved by the higher priority rx interrupt.

**Comment by cire831** [comment]
This bug is most likely from interrupt issues.  The gps int was at the same priority as the Radio FSM.  All interrupts are by default all at the same priority.

Fixed by a number of commits, including 6a689a2 and 803da4b.
